Streaming query documentation (#231)
AxonIQ/reference-guide: axon-server/performance/flow-control.md

* `axoniq.axonserver.queryFlowControl.nr-of-new-permits` \[5000\] - additional number of messages that the master can send to replica.
* `axoniq.axonserver.queryFlowControl.new-permits-threshold` \[5000\] - when replica reaches this threshold in remaining messages, it sends a request with additional number of messages to receive.

## Streaming query

> Flow control and stream cancellation features are only available with Axon Server 4.6 + version. Streaming queries used with pre Axon Server 4.6 version will work, but without these important features.
> Under the hood, backpressure does `Hop to Hop` signal propagation (see below) and inherits gRPC's [HTTP2-based backpressure model](https://developers.google.com/web/fundamentals/performance/http2/#flow_control).

Reviewer: ```suggestion
Flow control and stream cancellation features are only available with Axon Server 4.6.0 and up. 
When streaming queries are used with Axon Server versions before 4.6.0, it will work although without the following essential features.
Under the hood, backpressure does `Hop to Hop` signal propagation (see below) and inherits gRPC's [HTTP2-based backpressure model](https://developers.google.com/web/fundamentals/performance/http2/#flow_control).
```

> As a result, backpressure will not behave intuitively and will not propagate exact request signal from consumer to producer.
> HTTP/2 and Netty flow control has internal buffers based on message size, Axon Framework and Axon Server prefetch messages into internal buffers based on message count.
> Result is that the producer will send number of messages until it fills all the buffers, then backpressure will kick in.

Reviewer: ```suggestion
As a result, backpressure will not behave intuitively and will not propagate exact request signals from consumer to producer.
HTTP/2 and Netty flow control have internal buffers based on message size. 
In turn, Axon Framework and Axon Server prefetch messages into internal buffers based on message count.
The result is that the producer will send a number of messages until it fills all the buffers.
Only then will backpressure kick in.
```

> > **Hop to hop**
> >
> >Back-pressure signal is propagated per hop which is not an end to end connection,
> > which allows intermediate Axon Server to handle backpressure between two connections
> > and pre-fetch additional messages to increase overall performance.
> >

Reviewer: I am not entirely sure about my suggestion here, as I don't completely follow the intent of this notice.
So, if I make some wrong angles, be sure to fine-tune my adjustment.

```suggestion
> **Hop to hop**
>
> The backpressure signal is propagated per-hop.
> This approach makes it not an end-to-end connection that allows intermediate Axon Server instances to handle backpressure between two connections and pre-fetch additional messages to increase overall performance.
```

Reviewer: I also feel there's something missing from this notice, honestly.
Why add this specific? What does it tell the users?

> It's important to note that similar to backpressure, cancellation signal is per hop, meaning it's propagated over the network to Axon Server and then to producer, which can produce some latency in stream cancellation.
> Not to worry, any messages produced after consumer signaled cancellation will be ignored.

Reviewer: ```suggestion
It's important to note that similar to backpressure, the cancellation signal is also per hop.
This means it's propagated over the network to Axon Server and then to the producer.
This solution will thus introduce some latency in the stream cancellation.
Even though there is potential latency involved in cancellation, any messages produced **after** the consumer signaled cancellation will be ignored.
```
